File: TelerikUIExamples/TelerikUIExamples/ChartDocsGridCustomization.m
//
//  ChartDocsGridCustomization.m
//  TelerikUIExamples
//
//  Copyright © 2016 Telerik. All rights reserved.
//

#import "ChartDocsGridCustomization.h"
#import <TelerikUI/TelerikUI.h>

@implementation ChartDocsGridCustomization
{
    TKChart *chart;
}
- (void) viewDidLoad{
    [super viewDidLoad];
    
    chart = [[TKChart alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:chart];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    for (int i = 0; i<7; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:@(i+1) Y:@(arc4random() % 100)]];
    }
    
    TKChartLineSeries *lineSeries = [[TKChartLineSeries alloc] initWithItems:array];
    [chart addSeries:lineSeries];
    
}

- (void) zPosition
{
    // >> chart-grid-z
    TKChartGridStyle* gridStyle = chart.gridStyle;
    
    gridStyle.horizontalLineStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalLinesHidden = NO;
    gridStyle.horizontalFill = [TKSolidFill solidFillWithColor:[UIColor colorWithWhite:228.f/255.f alpha:0.7f]];
    gridStyle.horizontalAlternateFill = [TKSolidFill solidFillWithColor:[UIColor clearColor]];
    
    gridStyle.verticalFill = nil;
    gridStyle.verticalAlternateFill = nil;
    gridStyle.verticalLinesHidden = YES;
    
    gridStyle.zPosition = TKChartGridZPositionAboveSeries;
    // << chart-grid-z
}

- (void) combindingItAltogether
{
    // >> chart-grid-combining
    TKChartGridStyle* gridStyle = chart.gridStyle;
    
    gridStyle.horizontalLineStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalFill = [TKSolidFill solidFillWithColor:[UIColor colorWithWhite:228.f/255.f alpha:1.f]];
    gridStyle.horizontalAlternateFill = [TKSolidFill solidFillWithColor:[UIColor whiteColor]];
    gridStyle.horizontalLinesHidden = NO;
    
    gridStyle.verticalLineStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.verticalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.verticalLinesHidden = NO;
    gridStyle.verticalFill = [TKSolidFill solidFillWithColor:[UIColor colorWithRed:1.f green:1.f blue:0.f alpha:0.1f]];
    gridStyle.verticalAlternateFill = [TKSolidFill solidFillWithColor:[UIColor colorWithRed:0.f green:1.f blue:0.f alpha:0.1f]];
    // << chart-grid-combining
    
    // >> chart-grid-first
    gridStyle.drawOrder = TKChartGridDrawModeVerticalFirst;
    // << chart-grid-first
    
    // >> chart-grid-bg-fill
    gridStyle.backgroundFill = [TKSolidFill solidFillWithColor:[UIColor whiteColor]];
    // << chart-grid-bg-fill
    
    // >> chart-grid-img-fill
    gridStyle.backgroundFill = [TKSolidFill solidFillWithColor:
                                [UIColor colorWithPatternImage:[UIImage imageNamed:@"telerik_logo"]]];
    // << chart-grid-img-fill
    
}

- (void) colorfulGridLines
{
    // >> chart-grid-colorful
    TKChartGridStyle* gridStyle = chart.gridStyle;
    gridStyle.verticalLineStroke = [TKStroke strokeWithColor:[UIColor blackColor]];
    gridStyle.verticalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor blackColor]];
    gridStyle.verticalLinesHidden = NO;
    gridStyle.verticalFill = nil;
    gridStyle.verticalAlternateFill = nil;
    
    gridStyle.horizontalLineStroke = [TKStroke strokeWithColor:[UIColor redColor]];
    gridStyle.horizontalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor blueColor]];
    gridStyle.horizontalFill = nil;
    gridStyle.horizontalAlternateFill = nil;
    gridStyle.horizontalLinesHidden = NO;
    // << chart-grid-colorful
}

- (void) verticalAlternativeFills
{
    // >> chart-grid-alternate-vertical
    TKChartGridStyle* gridStyle = chart.gridStyle;
    
    gridStyle.verticalLineStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.verticalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.verticalLinesHidden = NO;
    gridStyle.verticalFill = [TKSolidFill solidFillWithColor:[UIColor colorWithWhite:228.f/255.f alpha:1.f]];
    gridStyle.verticalAlternateFill = [TKSolidFill solidFillWithColor:[UIColor whiteColor]];
    
    gridStyle.horizontalFill = nil;
    gridStyle.horizontalAlternateFill = nil;
    gridStyle.horizontalLinesHidden = YES;
    // << chart-grid-alternate-vertical
}

- (void) removeVerticalLines
{
    // >> chart-grid-remove-vertical
    TKChartGridStyle* gridStyle = chart.gridStyle;
    
    gridStyle.verticalLinesHidden = YES;
    gridStyle.verticalFill = nil;
    gridStyle.verticalAlternateFill = nil;
    
    gridStyle.horizontalLineStroke = [TKStroke strokeWithColor:[UIColor redColor]];
    gridStyle.horizontalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor blueColor]];
    gridStyle.horizontalFill = nil;
    gridStyle.horizontalAlternateFill = nil;
    gridStyle.horizontalLinesHidden = NO;
    // << chart-grid-remove-vertical
}

- (void) alternativeFills
{
    // >> chart-grid-alternate-horizontal
    TKChartGridStyle* gridStyle = chart.gridStyle;
    
    gridStyle.horizontalLineStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalLineAlternateStroke = [TKStroke strokeWithColor:[UIColor colorWithWhite:215.f/255.f alpha:1.f]];
    gridStyle.horizontalLinesHidden = NO;
    gridStyle.horizontalFill = [TKSolidFill solidFillWithColor:[UIColor colorWithWhite:228.f/255.f alpha:1.f]];
    gridStyle.horizontalAlternateFill = [TKSolidFill solidFillWithColor:[UIColor whiteColor]];
    
    gridStyle.verticalFill = nil;
    gridStyle.verticalAlternateFill = nil;
    gridStyle.verticalLinesHidden = YES;
    // << chart-grid-alternate-horizontal
}


@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ChartDocsGridCustomization.swift
//
//  ChartDocsGridCustomization.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

class ChartDocsGridCustomization: UIViewController {

    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        chart.frame = CGRectInset(self.view.bounds, 10, 10)
        self.view.addSubview(chart)
        
        let categories = ["Greetings", "Perfecto", "NearBy", "Family Store", "Fresh & Green" ]
        let expensesValues = [70, 75, 58, 59, 88]
        var expensesData = [TKChartDataPoint]()
        for i in 0 ..< categories.count {
            expensesData.append(TKChartDataPoint(x: categories[i], y: expensesValues[i]))
        }
        
        let series = TKChartLineSeries(items: expensesData)
        chart.addSeries(series)
    }
    
    func zPosition() {
        // >> chart-grid-z-swift
        let gridStyle = chart.gridStyle
        gridStyle.horizontalLineStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalLineAlternateStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalLinesHidden = false
        gridStyle.horizontalFill = TKSolidFill(color: UIColor(white: 228.0 / 255.0, alpha: 0.7))
        gridStyle.horizontalAlternateFill = TKSolidFill(color: UIColor.clearColor())
        
        gridStyle.verticalFill = nil
        gridStyle.verticalAlternateFill = nil
        gridStyle.verticalLinesHidden = true
        gridStyle.zPosition = TKChartGridZPosition.AboveSeries
        // << chart-grid-z-swift
    }
    
    func combiningItAltogether() {
        // >> chart-grid-combining-swift
        let gridStyle = chart.gridStyle
        gridStyle.horizontalLineStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalLineAlternateStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalFill = TKSolidFill(color: UIColor(white: 228.0 / 255.0, alpha: 0.7))
        gridStyle.horizontalAlternateFill = TKSolidFill(color: UIColor.whiteColor())
        gridStyle.horizontalLinesHidden = false
        
        gridStyle.verticalLineStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.verticalLineAlternateStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.verticalLinesHidden = false
        gridStyle.verticalFill = TKSolidFill(color: UIColor(red: 1.0, green: 1.0, blue:0.0 , alpha: 0.1))
        gridStyle.verticalAlternateFill = TKSolidFill(color: UIColor(red: 0.0, green: 1.0, blue: 0.0, alpha: 0.1))
        // << chart-grid-combining-swift
        
        // >> chart-grid-first-swift
        gridStyle.drawOrder = TKChartGridDrawMode.VerticalFirst
        // << chart-grid-first-swift
        
        // >> chart-grid-bg-fill-swift
        gridStyle.backgroundFill = TKSolidFill(color: UIColor.whiteColor())
        // << chart-grid-bg-fill-swift
        
        // >> chart-grid-img-fill-swift
        gridStyle.backgroundFill = TKSolidFill(color: UIColor(patternImage: UIImage(named: "logo")!))
        // << chart-grid-img-fill-swift
        
    }
    
    func colorfulGridLines() {
        // >> chart-grid-colorful-swift
        let gridStyle = chart.gridStyle
        gridStyle.verticalLineStroke = TKStroke(color: UIColor.blackColor())
        gridStyle.verticalLineAlternateStroke = TKStroke(color: UIColor.blackColor())
        gridStyle.verticalLinesHidden = false
        gridStyle.verticalFill = nil
        gridStyle.verticalAlternateFill = nil
        
        gridStyle.horizontalLineStroke = TKStroke(color: UIColor.redColor())
        gridStyle.horizontalLineAlternateStroke = TKStroke(color: UIColor.blueColor())
        gridStyle.horizontalFill = nil
        gridStyle.horizontalAlternateFill = nil
        gridStyle.horizontalLinesHidden = false
        // << chart-grid-colorful-swift
    }
    
    func verticalAlternativeFills() {
        // >> chart-grid-alternate-vertical-swift
        let gridStyle = chart.gridStyle
        gridStyle.verticalLineStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.verticalLineAlternateStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.verticalLinesHidden = false
        gridStyle.verticalFill = TKSolidFill(color: UIColor(white: 228.0 / 255.0, alpha: 1.0))
        gridStyle.verticalAlternateFill = TKSolidFill(color: UIColor.whiteColor())
        
        
        gridStyle.horizontalFill = nil
        gridStyle.horizontalAlternateFill = nil
        gridStyle.horizontalLinesHidden = true
        // << chart-grid-alternate-vertical-swift
    }
    
    func removeVerticalLines() {
        // >> chart-grid-remove-vertical-swift
        let gridStyle = chart.gridStyle
        gridStyle.verticalLinesHidden = true
        gridStyle.verticalFill = nil
        gridStyle.verticalAlternateFill = nil
        
        gridStyle.horizontalLineStroke = TKStroke(color: UIColor.redColor())
        gridStyle.horizontalLineAlternateStroke = TKStroke(color: UIColor.blueColor())
        gridStyle.horizontalFill = nil
        gridStyle.horizontalAlternateFill = nil
        gridStyle.horizontalLinesHidden = false
        // << chart-grid-remove-vertical-swift
    }
    
    func horizontalAlternativeFills() {
        // >> chart-grid-alternate-horizontal-swift
        let gridStyle = chart.gridStyle
        gridStyle.horizontalLineStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalLineAlternateStroke = TKStroke(color: UIColor(white: 215.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalLinesHidden = false
        gridStyle.horizontalFill = TKSolidFill(color: UIColor(white: 228.0 / 255.0, alpha: 1.0))
        gridStyle.horizontalAlternateFill = TKSolidFill(color: UIColor.whiteColor())
        
        gridStyle.verticalFill = nil
        gridStyle.verticalAlternateFill = nil
        gridStyle.verticalLinesHidden = true
        // << chart-grid-alternate-horizontal-swift
    }
}
